tests/boards/nrf52_bsim/egu/src/main.c

```diff
@@ -0,0 +1,505 @@
+	zassert_false(event_triggered_flag.triggered[6], "Event 6 has been triggered");
+	zassert_false(event_triggered_flag.triggered[7], "Event 7 has been triggered");
+	zassert_false(event_triggered_flag.triggered[8], "Event 8 has been triggered");
+	zassert_false(event_triggered_flag.triggered[9], "Event 9 has been triggered");
+	zassert_false(event_triggered_flag.triggered[10], "Event 10 has been triggered");
+	zassert_false(event_triggered_flag.triggered[11], "Event 11 has been triggered");
+	zassert_false(event_triggered_flag.triggered[12], "Event 12 has been triggered");
+	zassert_false(event_triggered_flag.triggered[13], "Event 13 has been triggered");
+	zassert_false(event_triggered_flag.triggered[14], "Event 14 has been triggered");
+	zassert_false(event_triggered_flag.triggered[15], "Event 15 has been triggered");
+
+	for (uint8_t i = 0; i < nrf_egu_channel_count(NRF_EGU5); i++) {
+		const nrf_egu_event_t check_event = nrf_egu_triggered_event_get(i);
+
+		zassert_false(nrf_egu_event_check(NRF_EGU5, check_event),
+				"event %d has been triggered, but it shouldn't", i);
+	}
+}
+
+static void timer_isr(const void *timer_isr_ctx)
+{
+	struct timer_isr_context *ctx = (struct timer_isr_context *)timer_isr_ctx;
+
+	if (ctx) {
+		nrf_egu_task_trigger(NRF_EGU5, nrf_egu_trigger_task_get(ctx->egu_channel));
+	}
+
+	nrf_timer_event_clear(TIMER_INSTANCE, NRF_TIMER_EVENT_COMPARE0);
+	nrf_timer_int_disable(TIMER_INSTANCE, TIMER_INT);
+}
+
+void test_nrf_egu_trigger_from_another_irq(void)
+{
+	uint32_t call_cnt_expected = event_triggered_flag.call_count + 1;
+	static struct timer_isr_context timer_isr_ctx = {
+		.egu_channel = 0
+	};
+
+	/* Timer cleanup */
+	nrf_timer_event_clear(TIMER_INSTANCE, NRF_TIMER_EVENT_COMPARE0);
+	nrf_timer_task_trigger(TIMER_INSTANCE, NRF_TIMER_TASK_STOP);
+	nrf_timer_task_trigger(TIMER_INSTANCE, NRF_TIMER_TASK_CLEAR);
+
+	/* Timer setup */
+	irq_connect_dynamic(TIMER_IRQ, TIMER_PRIORITY, timer_isr, &timer_isr_ctx, 0);
+	irq_enable(TIMER_IRQ);
+
+	nrf_timer_mode_set(TIMER_INSTANCE, NRF_TIMER_MODE_TIMER);
+	nrf_timer_bit_width_set(TIMER_INSTANCE, TIMER_BITMODE_BITMODE_16Bit);
+	nrf_timer_frequency_set(TIMER_INSTANCE, NRF_TIMER_FREQ_1MHz);
+	nrf_timer_cc_set(TIMER_INSTANCE, NRF_TIMER_CC_CHANNEL0, TIMER_DELAY_TICKS);
+	nrf_timer_int_enable(TIMER_INSTANCE, TIMER_INT);
+
+	/* EGU setup */
+	nrf_egu_int_enable(NRF_EGU5, NRF_EGU_INT_TRIGGERED0);
+	irq_connect_dynamic(SWI5_EGU5_IRQn, 0, SWI5_trigger_function, NULL, BIT(0));
+	irq_enable(SWI5_EGU5_IRQn);
+
+	nrf_timer_task_trigger(TIMER_INSTANCE, NRF_TIMER_TASK_START);
+	k_busy_wait(EGU_DELAY_USEC);
+
+	zassert_equal(call_cnt_expected, event_triggered_flag.call_count,
+			"interrupt called unexpected number of times %d",
+			event_triggered_flag.call_count);
+}
+
+
+static struct SWI_trigger_assert_parameter SWI4_event_triggered_flag;
+
+void SWI4_trigger_function(const void *param)
+{
+	SWI4_event_triggered_flag.call_count++;
```

Comment:
I don't understand what is purpose of this incremental.

Reply:
None for this ISR. Removed.